backlog/project.py

```diff
@@ -56,25 +148,87 @@ def get(self, projectIdOrKey):
+        """Update project
+        https://developer.nulab.com/docs/backlog/api/2/update-project/#form-parameters
+
+        :param projectIdOrKey: Project Name
+        :type projectIdOrKey: str
+        :param key: Project Key, defaults to None
+        :type key: str, optional
+        :param chartEnabled: Enable chart, defaults to None
+        :type chartEnabled: bool, optional
+        :param useResolvedForChart: Consider Resolved and statuses after as Closed, defaults to None
+        :type useResolvedForChart: bool, optional
+        :param subtaskingEnabled: Enable subtasking, defaults to None
+        :type subtaskingEnabled: _type_, optional
+        :param projectLeaderCanEditProjectLeader: Allow project administrators to manage each other, defaults to None
+        :type projectLeaderCanEditProjectLeader: bool, optional
+        :param useWiki: Enable Wiki, defaults to None
+        :type useWiki: bool, optional
+        :param useFileSharing: Enable shared files, defaults to None
+        :type useFileSharing: bool, optional
+        :param useWikiTreeView: Enable Wiki tree view, defaults to None
+        :type useWikiTreeView: bool, optional
+        :param useSubversion: Enable Subversion, defaults to None
+        :type useSubversion: bool, optional
+        :param useGit: Enable Git, defaults to None
+        :type useGit: bool, optional
+        :param useOriginalImageSizeAtWiki: Display images in Wikis in their original size, defaults to None
+        :type useOriginalImageSizeAtWiki: bool, optional
+        :param textFormattingRule: Formatting rules “backlog” or “markdown”, defaults to None
+        :type textFormattingRule: TextFormattingRule, optional
+        :param archived: Archive this project, defaults to None
+        :type archived: bool, optional
+        :param useDevAttributes: Enable priorities, versions and milestones, defaults to None
+        :type useDevAttributes: bool, optional
         """
-        raise NotImplementedError
+        _path = f"projects/{projectIdOrKey}"
+        _method = "PATCH"
+        request_param = {
+            "name": name,
+            "key": key,
+            "chartEnabled": chartEnabled,
+            "useResolvedForChart": useResolvedForChart,
+            "subtaskingEnabled": subtaskingEnabled,
+            "projectLeaderCanEditProjectLeader": projectLeaderCanEditProjectLeader,
+            "useWiki": useWiki,
+            "useFileSharing": useFileSharing,
+            "useWikiTreeView": useWikiTreeView,
+            "useSubversion": useSubversion,
+            "useGit": useGit,
+            "useOriginalImageSizeAtWiki": useOriginalImageSizeAtWiki,
+            "textFormattingRule": textFormattingRule,
+            "archived": archived,
+            "useDevAttributes": useDevAttributes
+        }
+        for k, v in request_param.copy().items():
+            if v is None:
+                del request_param[k]
+        if len(request_param.keys()) == 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)